drivers/dma/dma_mcux_lpc.c

```diff
@@ -312,13 +345,53 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 			data->descriptor_index = data->descriptor_used;
 		}
 	} else {
-		/* block_count shall be 1 */
-		/* Only one buffer, enable interrupt */
-		xferConfig = DMA_CHANNEL_XFER(0UL, 0UL, 1UL, 0UL,
-					config->dest_data_size,
-					src_inc,
-					dst_inc,
-					block_config->block_size);
+		/* If DMA Channel Chaining is expected, then the
+		 * burst length must be setup differently as well.
+		 */
+		if (!config->dest_chaining_en && !config->source_chaining_en) {
+			k_sem_give(&is_trig_mux_in_use);
```

Comment:
giving this for any channel causes a bug where if you:
set up A to link to B
set up B
set up B to link to C
in that order, then there will be a bug where channel A loses the mux unexpectedly